source/frontend/src/components/maps/hooks/useMapSearch.tsx

```diff
@@ -78,19 +78,26 @@ export const useMapSearch = () => {
         | Promise<FeatureCollection<Geometry, PMBC_FullyAttributed_Feature_Properties> | undefined>
         | undefined = undefined;
 
+      let findByPlanNumberTask:
+        | Promise<FeatureCollection<Geometry, PMBC_FullyAttributed_Feature_Properties> | undefined>
+        | undefined = undefined;
+
       loadPropertiesTask = pimsPropertyLayerService.loadPropertyLayer.execute(filter);
       if (filter?.PIN) {
         findByPinTask = fullyAttributedService.findByPin(filter?.PIN);
       }
       if (filter?.PID) {
         findByPidTask = fullyAttributedService.findByPid(filter?.PID);
       }
+      if (filter?.SURVEY_PLAN_NUMBER) {
+        findByPlanNumberTask = fullyAttributedService.findByPlanNumber(filter?.SURVEY_PLAN_NUMBER);
```

Comment:
this should be its own method to make sure the other filter parameters  (pid,pin)are not trumping this. Also, I dont think it would be possible to do a pid,pin + survey plan search

Reply:
agreed pid/pin is a simultaneous search. Ie. a single search field that can be interpreted two ways. not the case here.

Reply:
Fixed along with some minor refactoring of the searchByMany method code due to a lot of repetition.